**Linux** [Build 9988](https://ci.wildfly.org/viewLog.html?buildId=22021&buildTypeId=WFPR) outcome was **FAILURE** using a merge of 537aa72246dbe5fdd09b1713c3151a33143e4b1d
Summary: Tests failed: 21 (21 new), passed: 3409, ignored: 193 Build time: 01:03:32
### Failed tests
```
org.jboss.as.test.integration.domain.mixed.eap640.RBACConfig640TestCase.testAddHostScopedRole:     "operation" => "read-attribute",
    "address" => [
        ("core-service" => "management"),
        ("access" => "authorization"),
        ("host-scoped-role" => "WFCORE-1622_H")
    ],
    "name" => "base-role"
}
{
    "outcome" => "failed",
    "failure-description" => "JBAS014807: Management resource '[
    (\"core-service\" => \"management\"),
    (\"access\" => \"authorization\"),
    (\"host-scoped-role\" => \"WFCORE-1622_H\")
]' not found",
    "rolled-back" => true
} expected:<[success]> but was:<[failed]>
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.executeForResult(RBACConfigTestCase.java:245)
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.addTest(RBACConfigTestCase.java:222)
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.testAddHostScopedRole(RBACConfigTestCase.java:113)
------- Stdout: -------
[Host Controller]  [33m [0m [33m13:19:30,290 WARN  [org.jboss.as.protocol.connection] (Remoting "build26-el7:MANAGEMENT" task-2) JBAS012118: No such request (31) associated with channel Channel ID fa2a27b0 (outbound) of Remoting connection 01d06af8 to localhost/0:0:0:0:0:0:0:1:9999 [0m
[Host Controller]  [33m [0m [33m13:19:30,295 WARN  [org.jboss.as.protocol.connection] (management task-9) WFLYPRT0018: No such request (31) associated with channel Channel ID 7a2a27b0 (inbound) of Remoting connection 00672278 to /0:0:0:0:0:0:0:1:52750 of endpoint "master:MANAGEMENT" <165a310> [0m
[Host Controller]  [33m [0m [0m13:19:30,323 INFO  [org.jboss.as.domain.controller] (management task-10) WFLYHC0026: Unregistered remote slave host "slave" [0m
 [0m13:19:32,406 INFO  [org.jboss.modules] (main) JBoss Modules version 1.5.2.Final
 [0m [0m13:19:32,568 INFO  [org.jboss.as.process.Host Controller.status] (main) WFLYPC0018: Starting process 'Host Controller'
 [0m[Host Controller]  [0m13:19:33,021 INFO  [org.jboss.modules] (main) JBoss Modules version 1.5.2.Final [0m
[Host Controller]  [0m [0m13:19:33,180 INFO  [org.jboss.msc] (main) JBoss MSC version 1.2.6.Final [0m
[Host Controller]  [0m [0m13:19:33,243 INFO  [org.jboss.as] (MSC service thread 1-7) WFLYSRV0049: WildFly Full 10.1.0.Final-SNAPSHOT (WildFly Core 2.2.0.CR8) starting [0m
[Host Controller]  [0m [0m13:19:34,066 INFO  [org.jboss.as.controller.management-deprecated] (Controller Boot Thread) WFLYCTL0028: Attribute 'permgen-size' in the resource at address '/host=master/jvm=default' is deprecated, and may be removed in future version. See the attribute description in the output of the read-resource-description operation to learn more about the deprecation. [0m
[Host Controller]  [0m [0m13:19:34,066 INFO  [org.jboss.as.controller.management-deprecated] (Controller Boot Thread) WFLYCTL0028: Attribute 'max-permgen-size' in the resource at address '/host=master/jvm=default' is deprecated, and may be removed in future version. See the attribute description in the output of the read-resource-description operation to learn more about the deprecation. [0m
[Host Controller]  [0m [0m13:19:34,104 INFO  [org.jboss.as.host.controller] (Controller Boot Thread) WFLYHC0003: Creating http management service using network interface (management) port (9990) securePort (-1) [0m
[Host Controller]  [0m [0m13:19:34,107 INFO  [org.xnio] (MSC service thread 1-4) XNIO version 3.4.0.Beta3 [0m
[Host Controller]  [0m [0m13:19:34,112 INFO  [org.xnio.nio] (MSC service thread 1-4) XNIO NIO Implementation Version 3.4.0.Beta3 [0m
[Host Controller]  [0m [33m13:19:34,151 WARN  [org.jboss.as.domain.management.security] (MSC service thread 1-2) WFLYDM0111: Keystore /opt/buildAgent/work/a31d203e70e89f90/testsuite/mixed-domain/target/domains/KernelBehavior640TestSuite/master/configuration/application.keystore not found, it will be auto generated on first use with a self signed certificate for host localhost [0m
[Host Controller]  [33m [0m [0m13:19:34,197 INFO  [org.jboss.remoting] (MSC service thread 1-4) JBoss Remoting version 4.0.21.Final [0m
[Host Controller]  [0m [0m13:19:34,222 INFO  [org.jboss.as.remoting] (MSC service thread 1-4) WFLYRMT0001: Listening on [::1]:9999 [0m
[Host Controller]  [0m [0m13:19:34,462 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0060: Http management interface listening on http://[::1]:9990/management [0m
[Host Controller]  [0m [0m13:19:34,463 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0051: Admin console listening on http://[::1]:9990 [0m
[Host Controller]  [0m [0m13:19:34,463 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0025: WildFly Full 10.1.0.Final-SNAPSHOT (WildFly Core 2.2.0.CR8) (Host Controller) started in 1794ms - Started 48 of 52 services (18 services are lazy, passive or on-demand) [0m
INFO  [org.jboss.as.test.integration.domain.management.util.DomainLifecycleUtil] All servers started in 2481 ms
INFO  [org.jboss.as.test.integration.domain.management.util.DomainLifecycleUtil] HostController started in 2739 ms
 [0m13:19:35,224 INFO  [org.jboss.modules] (main) JBoss Modules version 1.3.6.Final-redhat-1
 [0m [0m13:19:35,404 INFO  [org.jboss.as.process.Host Controller.status] (main) JBAS012017: Starting process 'Host Controller'
 [0m[Host Controller]  [0m13:19:36,319 INFO  [org.jboss.modules] (main) JBoss Modules version 1.3.6.Final-redhat-1 [0m
[Host Controller]  [0m [0m13:19:36,570 INFO  [org.jboss.msc] (main) JBoss MSC version 1.1.5.Final-redhat-1 [0m
[Host Controller]  [0m [0m13:19:36,657 INFO  [org.jboss.as] (MSC service thread 1-7) JBAS015899: JBoss EAP 6.4.0.GA (AS 7.5.0.Final-redhat-21) starting [0m
[Host Controller]  [0m [0m13:19:37,322 INFO  [org.xnio] (MSC service thread 1-1) XNIO Version 3.0.13.GA-redhat-1 [0m
[Host Controller]  [0m [0m13:19:37,325 INFO  [org.xnio.nio] (MSC service thread 1-1) XNIO NIO Implementation Version 3.0.13.GA-redhat-1 [0m
[Host Controller]  [0m [0m13:19:37,340 INFO  [org.jboss.remoting] (MSC service thread 1-1) JBoss Remoting version 3.3.4.Final-redhat-1 [0m
[Host Controller]  [0m [0m13:19:37,371 INFO  [org.jboss.as.remoting] (MSC service thread 1-6) JBAS017100: Listening on [::1]:19999 [0m


org.jboss.as.test.integration.domain.mixed.eap640.RBACConfig640TestCase.testModifyServerSSLSensitivityConstraint:     "operation" => "read-attribute",
    "address" => [
        ("core-service" => "management"),
        ("access" => "authorization"),
        ("constraint" => "sensitivity-classification"),
        ("type" => "core"),
        ("classification" => "server-ssl")
    ],
    "name" => "configured-requires-write"
}
{
    "outcome" => "failed",
    "failure-description" => "JBAS014807: Management resource '[
    (\"core-service\" => \"management\"),
    (\"access\" => \"authorization\"),
    (\"constraint\" => \"sensitivity-classification\"),
    (\"type\" => \"core\"),
    (\"classification\" => \"server-ssl\")
]' not found",
    "rolled-back" => true
} expected:<[success]> but was:<[failed]>
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.executeForResult(RBACConfigTestCase.java:245)
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.modifyTest(RBACConfigTestCase.java:182)
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.testModifyServerSSLSensitivityConstraint(RBACConfigTestCase.java:143)


org.jboss.as.test.integration.domain.mixed.eap640.RBACConfig640TestCase.testAddServerGroupScopedRole:     "operation" => "read-attribute",
    "address" => [
        ("core-service" => "management"),
        ("access" => "authorization"),
        ("server-group-scoped-role" => "WFCORE-1622_S")
    ],
    "name" => "base-role"
}
{
    "outcome" => "failed",
    "failure-description" => "JBAS014807: Management resource '[
    (\"core-service\" => \"management\"),
    (\"access\" => \"authorization\"),
    (\"server-group-scoped-role\" => \"WFCORE-1622_S\")
]' not found",
    "rolled-back" => true
} expected:<[success]> but was:<[failed]>
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.executeForResult(RBACConfigTestCase.java:245)
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.addTest(RBACConfigTestCase.java:222)
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.testAddServerGroupScopedRole(RBACConfigTestCase.java:125)
------- Stdout: -------
[Host Controller]  [33m [0m [33m13:19:40,877 WARN  [org.jboss.as.protocol.connection] (Remoting "build26-el7:MANAGEMENT" task-7) JBAS012118: No such request (10) associated with channel Channel ID e2ffc3af (outbound) of Remoting connection 00f5fce7 to localhost/0:0:0:0:0:0:0:1:9999 [0m
[Host Controller]  [33m [0m [33m13:19:40,877 WARN  [org.jboss.as.protocol.connection] (management task-4) WFLYPRT0018: No such request (10) associated with channel Channel ID 62ffc3af (inbound) of Remoting connection 003785fd to /0:0:0:0:0:0:0:1:51905 of endpoint "master:MANAGEMENT" <1350c9f> [0m


org.jboss.as.test.integration.domain.mixed.eap640.RBACConfig640TestCase.testModifySensitivityConstraint: java.lang.AssertionError: expected:<true> but was:<undefined>
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.modifyTest(RBACConfigTestCase.java:194)
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.testModifySensitivityConstraint(RBACConfigTestCase.java:134)
------- Stdout: -------
[Host Controller]  [33m [0m [33m13:19:40,920 WARN  [org.jboss.as.protocol.connection] (management task-7) WFLYPRT0018: No such request (11) associated with channel Channel ID 62ffc3af (inbound) of Remoting connection 003785fd to /0:0:0:0:0:0:0:1:51905 of endpoint "master:MANAGEMENT" <1350c9f> [0m
[Host Controller]  [33m [0m [33m13:19:40,920 WARN  [org.jboss.as.protocol.connection] (Remoting "build26-el7:MANAGEMENT" task-9) JBAS012118: No such request (11) associated with channel Channel ID e2ffc3af (outbound) of Remoting connection 00f5fce7 to localhost/0:0:0:0:0:0:0:1:9999 [0m
[Host Controller]  [33m [0m [33m13:19:40,986 WARN  [org.jboss.as.protocol.connection] (management task-3) WFLYPRT0018: No such request (14) associated with channel Channel ID 62ffc3af (inbound) of Remoting connection 003785fd to /0:0:0:0:0:0:0:1:51905 of endpoint "master:MANAGEMENT" <1350c9f> [0m
[Host Controller]  [33m [0m [33m13:19:40,987 WARN  [org.jboss.as.protocol.connection] (Remoting "build26-el7:MANAGEMENT" task-1) JBAS012118: No such request (14) associated with channel Channel ID e2ffc3af (outbound) of Remoting connection 00f5fce7 to localhost/0:0:0:0:0:0:0:1:9999 [0m
[Host Controller]  [33m [0m [33m13:19:41,007 WARN  [org.jboss.as.protocol.connection] (Remoting "build26-el7:MANAGEMENT" task-4) JBAS012118: No such request (15) associated with channel Channel ID e2ffc3af (outbound) of Remoting connection 00f5fce7 to localhost/0:0:0:0:0:0:0:1:9999 [0m


org.jboss.as.test.integration.domain.mixed.eap640.RBACConfig640TestCase.testAddRoleMapping:     "operation" => "read-attribute",
    "address" => [
        ("core-service" => "management"),
        ("access" => "authorization"),
        ("role-mapping" => "Operator")
    ],
    "name" => "include-all"
}
{
    "outcome" => "failed",
    "failure-description" => "JBAS014807: Management resource '[
    (\"core-service\" => \"management\"),
    (\"access\" => \"authorization\"),
    (\"role-mapping\" => \"Operator\")
]' not found",
    "rolled-back" => true
} expected:<[success]> but was:<[failed]>
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.executeForResult(RBACConfigTestCase.java:245)
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.addTest(RBACConfigTestCase.java:222)
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.testAddRoleMapping(RBACConfigTestCase.java:101)
------- Stdout: -------
[Host Controller]  [33m [0m [33m13:19:41,117 WARN  [org.jboss.as.protocol.connection] (Remoting "build26-el7:MANAGEMENT" task-3) JBAS012118: No such request (21) associated with channel Channel ID e2ffc3af (outbound) of Remoting connection 00f5fce7 to localhost/0:0:0:0:0:0:0:1:9999 [0m
[Host Controller]  [33m [0m [33m13:19:41,120 WARN  [org.jboss.as.protocol.connection] (management task-8) WFLYPRT0018: No such request (21) associated with channel Channel ID 62ffc3af (inbound) of Remoting connection 003785fd to /0:0:0:0:0:0:0:1:51905 of endpoint "master:MANAGEMENT" <1350c9f> [0m


org.jboss.as.test.integration.domain.mixed.eap640.RBACConfig640TestCase.testModifySensitiveExpressionsConstraint: java.lang.AssertionError: expected:<true> but was:<undefined>
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.modifyTest(RBACConfigTestCase.java:194)
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.testModifySensitiveExpressionsConstraint(RBACConfigTestCase.java:159)
------- Stdout: -------
[Host Controller]  [33m [0m [33m13:19:41,144 WARN  [org.jboss.as.protocol.connection] (Remoting "build26-el7:MANAGEMENT" task-2) JBAS012118: No such request (22) associated with channel Channel ID e2ffc3af (outbound) of Remoting connection 00f5fce7 to localhost/0:0:0:0:0:0:0:1:9999 [0m
[Host Controller]  [33m [0m [33m13:19:41,154 WARN  [org.jboss.as.protocol.connection] (management task-1) WFLYPRT0018: No such request (22) associated with channel Channel ID 62ffc3af (inbound) of Remoting connection 003785fd to /0:0:0:0:0:0:0:1:51905 of endpoint "master:MANAGEMENT" <1350c9f> [0m
[Host Controller]  [33m [0m [33m13:19:41,179 WARN  [org.jboss.as.protocol.connection] (Remoting "build26-el7:MANAGEMENT" task-11) JBAS012118: No such request (25) associated with channel Channel ID e2ffc3af (outbound) of Remoting connection 00f5fce7 to localhost/0:0:0:0:0:0:0:1:9999 [0m
[Host Controller]  [33m [0m [33m13:19:41,180 WARN  [org.jboss.as.protocol.connection] (management task-6) WFLYPRT0018: No such request (25) associated with channel Channel ID 62ffc3af (inbound) of Remoting connection 003785fd to /0:0:0:0:0:0:0:1:51905 of endpoint "master:MANAGEMENT" <1350c9f> [0m
[Host Controller]  [33m [0m [33m13:19:41,197 WARN  [org.jboss.as.protocol.connection] (Remoting "build26-el7:MANAGEMENT" task-14) JBAS012118: No such request (26) associated with channel Channel ID e2ffc3af (outbound) of Remoting connection 00f5fce7 to localhost/0:0:0:0:0:0:0:1:9999 [0m
[Host Controller]  [33m [0m [33m13:19:41,197 WARN  [org.jboss.as.protocol.connection] (management task-10) WFLYPRT0018: No such request (26) associated with channel Channel ID 62ffc3af (inbound) of Remoting connection 003785fd to /0:0:0:0:0:0:0:1:51905 of endpoint "master:MANAGEMENT" <1350c9f> [0m


org.jboss.as.test.integration.domain.mixed.eap640.RBACConfig640TestCase.testModifyApplicationConstraint: java.lang.AssertionError: expected:<true> but was:<undefined>
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.modifyTest(RBACConfigTestCase.java:194)
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.testModifyApplicationConstraint(RBACConfigTestCase.java:152)
------- Stdout: -------
[Host Controller]  [33m [0m [33m13:19:41,244 WARN  [org.jboss.as.protocol.connection] (Remoting "build26-el7:MANAGEMENT" task-5) JBAS012118: No such request (29) associated with channel Channel ID e2ffc3af (outbound) of Remoting connection 00f5fce7 to localhost/0:0:0:0:0:0:0:1:9999 [0m
[Host Controller]  [33m [0m [33m13:19:41,250 WARN  [org.jboss.as.protocol.connection] (management task-7) WFLYPRT0018: No such request (29) associated with channel Channel ID 62ffc3af (inbound) of Remoting connection 003785fd to /0:0:0:0:0:0:0:1:51905 of endpoint "master:MANAGEMENT" <1350c9f> [0m
[Host Controller]  [33m [0m [33m13:19:41,269 WARN  [org.jboss.as.protocol.connection] (Remoting "build26-el7:MANAGEMENT" task-8) JBAS012118: No such request (30) associated with channel Channel ID e2ffc3af (outbound) of Remoting connection 00f5fce7 to localhost/0:0:0:0:0:0:0:1:9999 [0m
[Host Controller]  [33m [0m [33m13:19:41,269 WARN  [org.jboss.as.protocol.connection] (management task-9) WFLYPRT0018: No such request (30) associated with channel Channel ID 62ffc3af (inbound) of Remoting connection 003785fd to /0:0:0:0:0:0:0:1:51905 of endpoint "master:MANAGEMENT" <1350c9f> [0m


org.jboss.as.test.integration.domain.mixed.eap630.RBACConfig630TestCase.testAddHostScopedRole:     "operation" => "read-attribute",
    "address" => [
        ("core-service" => "management"),
        ("access" => "authorization"),
        ("host-scoped-role" => "WFCORE-1622_H")
    ],
    "name" => "base-role"
}
{
    "outcome" => "failed",
    "failure-description" => "JBAS014807: Management resource '[
    (\"core-service\" => \"management\"),
    (\"access\" => \"authorization\"),
    (\"host-scoped-role\" => \"WFCORE-1622_H\")
]' not found",
    "rolled-back" => true
} expected:<[success]> but was:<[failed]>
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.executeForResult(RBACConfigTestCase.java:245)
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.addTest(RBACConfigTestCase.java:222)
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.testAddHostScopedRole(RBACConfigTestCase.java:113)
------- Stdout: -------
[Host Controller]  [33m [0m [33m13:24:14,888 WARN  [org.jboss.as.protocol.connection] (Remoting "build26-el7:MANAGEMENT" task-12) JBAS012118: No such request (35) associated with channel Channel ID c27f2a5e (outbound) of Remoting connection 019ef3e2 to localhost/0:0:0:0:0:0:0:1:9999 [0m
[Host Controller]  [33m [0m [33m13:24:14,894 WARN  [org.jboss.as.protocol.connection] (management task-2) WFLYPRT0018: No such request (35) associated with channel Channel ID 427f2a5e (inbound) of Remoting connection 002d3cd5 to /0:0:0:0:0:0:0:1:54021 of endpoint "master:MANAGEMENT" <1cd270a> [0m
[Host Controller]  [33m [0m [0m13:24:14,916 INFO  [org.jboss.as.domain.controller] (management task-4) WFLYHC0026: Unregistered remote slave host "slave" [0m
 [0m13:24:16,805 INFO  [org.jboss.modules] (main) JBoss Modules version 1.5.2.Final
 [0m [0m13:24:16,939 INFO  [org.jboss.as.process.Host Controller.status] (main) WFLYPC0018: Starting process 'Host Controller'
 [0m[Host Controller]  [0m13:24:17,597 INFO  [org.jboss.modules] (main) JBoss Modules version 1.5.2.Final [0m
[Host Controller]  [0m [0m13:24:17,759 INFO  [org.jboss.msc] (main) JBoss MSC version 1.2.6.Final [0m
[Host Controller]  [0m [0m13:24:17,816 INFO  [org.jboss.as] (MSC service thread 1-7) WFLYSRV0049: WildFly Full 10.1.0.Final-SNAPSHOT (WildFly Core 2.2.0.CR8) starting [0m
[Host Controller]  [0m [0m13:24:18,425 INFO  [org.jboss.as.controller.management-deprecated] (Controller Boot Thread) WFLYCTL0028: Attribute 'permgen-size' in the resource at address '/host=master/jvm=default' is deprecated, and may be removed in future version. See the attribute description in the output of the read-resource-description operation to learn more about the deprecation. [0m
[Host Controller]  [0m [0m13:24:18,425 INFO  [org.jboss.as.controller.management-deprecated] (Controller Boot Thread) WFLYCTL0028: Attribute 'max-permgen-size' in the resource at address '/host=master/jvm=default' is deprecated, and may be removed in future version. See the attribute description in the output of the read-resource-description operation to learn more about the deprecation. [0m
[Host Controller]  [0m [0m13:24:18,456 INFO  [org.jboss.as.host.controller] (Controller Boot Thread) WFLYHC0003: Creating http management service using network interface (management) port (9990) securePort (-1) [0m
[Host Controller]  [0m [0m13:24:18,459 INFO  [org.xnio] (MSC service thread 1-3) XNIO version 3.4.0.Beta3 [0m
[Host Controller]  [0m [0m13:24:18,465 INFO  [org.xnio.nio] (MSC service thread 1-3) XNIO NIO Implementation Version 3.4.0.Beta3 [0m
[Host Controller]  [0m [33m13:24:18,494 WARN  [org.jboss.as.domain.management.security] (MSC service thread 1-6) WFLYDM0111: Keystore /opt/buildAgent/work/a31d203e70e89f90/testsuite/mixed-domain/target/domains/KernelBehavior630TestSuite/master/configuration/application.keystore not found, it will be auto generated on first use with a self signed certificate for host localhost [0m
[Host Controller]  [33m [0m [0m13:24:18,504 INFO  [org.jboss.remoting] (MSC service thread 1-3) JBoss Remoting version 4.0.21.Final [0m
[Host Controller]  [0m [0m13:24:18,550 INFO  [org.jboss.as.remoting] (MSC service thread 1-3) WFLYRMT0001: Listening on [::1]:9999 [0m
[Host Controller]  [0m [0m13:24:18,778 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0060: Http management interface listening on http://[::1]:9990/management [0m
[Host Controller]  [0m [0m13:24:18,778 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0051: Admin console listening on http://[::1]:9990 [0m
[Host Controller]  [0m [0m13:24:18,778 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0025: WildFly Full 10.1.0.Final-SNAPSHOT (WildFly Core 2.2.0.CR8) (Host Controller) started in 1682ms - Started 48 of 52 services (18 services are lazy, passive or on-demand) [0m
INFO  [org.jboss.as.test.integration.domain.management.util.DomainLifecycleUtil] All servers started in 2402 ms
INFO  [org.jboss.as.test.integration.domain.management.util.DomainLifecycleUtil] HostController started in 2658 ms
 [0m13:24:19,502 INFO  [org.jboss.modules] (main) JBoss Modules version 1.3.3.Final-redhat-1
 [0m [0m13:24:19,606 INFO  [org.jboss.as.process.Host Controller.status] (main) JBAS012017: Starting process 'Host Controller'
 [0m[Host Controller]  [0m13:24:20,347 INFO  [org.jboss.modules] (main) JBoss Modules version 1.3.3.Final-redhat-1 [0m
[Host Controller]  [0m [0m13:24:20,525 INFO  [org.jboss.msc] (main) JBoss MSC version 1.1.5.Final-redhat-1 [0m
[Host Controller]  [0m [0m13:24:20,576 INFO  [org.jboss.as] (MSC service thread 1-6) JBAS015899: JBoss EAP 6.3.0.GA (AS 7.4.0.Final-redhat-19) starting [0m
[Host Controller]  [0m [0m13:24:21,359 INFO  [org.xnio] (MSC service thread 1-3) XNIO Version 3.0.10.GA-redhat-1 [0m
[Host Controller]  [0m [0m13:24:21,366 INFO  [org.xnio.nio] (MSC service thread 1-3) XNIO NIO Implementation Version 3.0.10.GA-redhat-1 [0m
[Host Controller]  [0m [0m13:24:21,405 INFO  [org.jboss.remoting] (MSC service thread 1-3) JBoss Remoting version (unknown) [0m
[Host Controller]  [0m [0m13:24:21,459 INFO  [org.jboss.as.remoting] (MSC service thread 1-1) JBAS017100: Listening on [::1]:19999 [0m


org.jboss.as.test.integration.domain.mixed.eap630.RBACConfig630TestCase.testModifyServerSSLSensitivityConstraint:     "operation" => "read-attribute",
    "address" => [
        ("core-service" => "management"),
        ("access" => "authorization"),
        ("constraint" => "sensitivity-classification"),
        ("type" => "core"),
        ("classification" => "server-ssl")
    ],
    "name" => "configured-requires-write"
}
{
    "outcome" => "failed",
    "failure-description" => "JBAS014807: Management resource '[
    (\"core-service\" => \"management\"),
    (\"access\" => \"authorization\"),
    (\"constraint\" => \"sensitivity-classification\"),
    (\"type\" => \"core\"),
    (\"classification\" => \"server-ssl\")
]' not found",
    "rolled-back" => true
} expected:<[success]> but was:<[failed]>
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.executeForResult(RBACConfigTestCase.java:245)
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.modifyTest(RBACConfigTestCase.java:182)
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.testModifyServerSSLSensitivityConstraint(RBACConfigTestCase.java:143)


org.jboss.as.test.integration.domain.mixed.eap630.RBACConfig630TestCase.testAddServerGroupScopedRole:     "operation" => "read-attribute",
    "address" => [
        ("core-service" => "management"),
        ("access" => "authorization"),
        ("server-group-scoped-role" => "WFCORE-1622_S")
    ],
    "name" => "base-role"
}
{
    "outcome" => "failed",
    "failure-description" => "JBAS014807: Management resource '[
    (\"core-service\" => \"management\"),
    (\"access\" => \"authorization\"),
    (\"server-group-scoped-role\" => \"WFCORE-1622_S\")
]' not found",
    "rolled-back" => true
} expected:<[success]> but was:<[failed]>
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.executeForResult(RBACConfigTestCase.java:245)
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.addTest(RBACConfigTestCase.java:222)
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.testAddServerGroupScopedRole(RBACConfigTestCase.java:125)
------- Stdout: -------
[Host Controller]  [33m [0m [33m13:24:24,811 WARN  [org.jboss.as.protocol.connection] (Remoting "build26-el7:MANAGEMENT" task-7) JBAS012118: No such request (11) associated with channel Channel ID 812bce33 (outbound) of Remoting connection 01f6ca9b to localhost/0:0:0:0:0:0:0:1:9999 [0m
[Host Controller]  [33m [0m [33m13:24:24,812 WARN  [org.jboss.as.protocol.connection] (management task-5) WFLYPRT0018: No such request (11) associated with channel Channel ID 012bce33 (inbound) of Remoting connection 0133c0c2 to /0:0:0:0:0:0:0:1:43776 of endpoint "master:MANAGEMENT" <1f43654> [0m
[Host Controller]  [33m [0m [33m13:24:24,841 WARN  [org.jboss.as.protocol.connection] (Remoting "build26-el7:MANAGEMENT" task-10) JBAS012118: No such request (12) associated with channel Channel ID 812bce33 (outbound) of Remoting connection 01f6ca9b to localhost/0:0:0:0:0:0:0:1:9999 [0m
[Host Controller]  [33m [0m [33m13:24:24,841 WARN  [org.jboss.as.protocol.connection] (management task-8) WFLYPRT0018: No such request (12) associated with channel Channel ID 012bce33 (inbound) of Remoting connection 0133c0c2 to /0:0:0:0:0:0:0:1:43776 of endpoint "master:MANAGEMENT" <1f43654> [0m


org.jboss.as.test.integration.domain.mixed.eap630.RBACConfig630TestCase.testModifySensitivityConstraint: java.lang.AssertionError: expected:<true> but was:<undefined>
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.modifyTest(RBACConfigTestCase.java:194)
	at org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase.testModifySensitivityConstraint(RBACConfigTestCase.java:134)
------- Stdout: -------
[Host Controller]  [33m [0m [33m13:24:24,894 WARN  [org.jboss.as.protocol.connection] (management task-4) WFLYPRT0018: No such request (15) associated with channel Channel ID 012bce33 (inbound) of Remoting connection 0133c0c2 to /0:0:0:0:0:0:0:1:43776 of endpoint "master:MANAGEMENT" <1f43654> [0m
[Host Controller]  [33m [0m [33m13:24:24,895 WARN  [org.jboss.as.protocol.connection] (Remoting "build26-el7:MANAGEMENT" task-2) JBAS012118: No such request (15) associated with channel Channel ID 812bce33 (outbound) of Remoting connection 01f6ca9b to localhost/0:0:0:0:0:0:0:1:9999 [0m



##### there are 11 more failed tests, see build details
```

--- a/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/RBACConfigTestCase.java
+++ b/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/RBACConfigTestCase.java
@@ -26,6 +26,7 @@ import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.CON
 import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.CONSTRAINT;
 import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.CORE;
 import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.CORE_SERVICE;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.FAILED;
 import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.HOST;
 import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.HOSTS;
 import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.HOST_SCOPED_ROLE;
@@ -131,7 +132,7 @@ public class RBACConfigTestCase {
         PathAddress mapping = RBAC_BASE.append(CONSTRAINT, SENSITIVITY_CLASSIFICATION)
                 .append(TYPE, CORE)
                 .append(CLASSIFICATION, "socket-config");
-        modifyTest(mapping, CONFIGURED_REQUIRES_WRITE, new ModelNode(true));
+        modifyTest(mapping, CONFIGURED_REQUIRES_WRITE, new ModelNode(true), true);
     }
 
     @Test
@@ -140,7 +141,7 @@ public class RBACConfigTestCase {
         PathAddress mapping = RBAC_BASE.append(CONSTRAINT, SENSITIVITY_CLASSIFICATION)
                 .append(TYPE, CORE)
                 .append(CLASSIFICATION, "server-ssl");
-        modifyTest(mapping, CONFIGURED_REQUIRES_WRITE, new ModelNode(true));
+        modifyTest(mapping, CONFIGURED_REQUIRES_WRITE, new ModelNode(true), getSupportsServerSSL());
     }
 
     @Test
@@ -149,14 +150,19 @@ public class RBACConfigTestCase {
         PathAddress mapping = RBAC_BASE.append(CONSTRAINT, APPLICATION_CLASSIFICATION)
                 .append(TYPE, CORE)
                 .append(CLASSIFICATION, "deployment");
-        modifyTest(mapping, CONFIGURED_APPLICATION, new ModelNode(true));
+        modifyTest(mapping, CONFIGURED_APPLICATION, new ModelNode(true), true);
     }
 
     @Test
     public void testModifySensitiveExpressionsConstraint() throws IOException {
 
         PathAddress mapping = RBAC_BASE.append(CONSTRAINT, VAULT_EXPRESSION);
-        modifyTest(mapping, CONFIGURED_REQUIRES_WRITE, new ModelNode(true));
+        modifyTest(mapping, CONFIGURED_REQUIRES_WRITE, new ModelNode(true), true);
+    }
+
+    /** Override this to return false in subclasses that test EAP < 6.4.7 */
+    protected boolean getSupportsServerSSL() {
+        return true;
     }
 
     private void modifyTest(PathAddress base, String attribute) throws IOException {
@@ -177,12 +183,17 @@ public class RBACConfigTestCase {
         assertEquals(masterValue, serverValue);
     }
 
-    private void modifyTest(PathAddress base, String attribute, ModelNode newValue) throws IOException {
+    private void modifyTest(PathAddress base, String attribute, ModelNode newValue, boolean expectSlaveEffect) throws IOException {
         ModelNode masterValue = executeForResult(Util.getReadAttributeOperation(base, attribute), masterClient);
-        ModelNode slaveValue = executeForResult(Util.getReadAttributeOperation(base, attribute), slaveClient);
-        assertEquals(masterValue, slaveValue);
-        ModelNode serverValue = executeForResult(Util.getReadAttributeOperation(SERVER_ONE.append(base), attribute), masterClient);
-        assertEquals(masterValue, serverValue);
+        if (expectSlaveEffect) {
+            ModelNode slaveValue = executeForResult(Util.getReadAttributeOperation(base, attribute), slaveClient);
+            assertEquals(masterValue, slaveValue);
+            ModelNode serverValue = executeForResult(Util.getReadAttributeOperation(SERVER_ONE.append(base), attribute), masterClient);
+            assertEquals(masterValue, serverValue);
+        } else {
+            executeForFailure(Util.getReadAttributeOperation(base, attribute), slaveClient);
+            executeForFailure(Util.getReadAttributeOperation(SERVER_ONE.append(base), attribute), masterClient);
+        }
 
         // Write the same value, so we don't need to clean up.
         Throwable caught = null;
@@ -190,10 +201,15 @@ public class RBACConfigTestCase {
         try {
             ModelNode newMasterValue = executeForResult(Util.getReadAttributeOperation(base, attribute), masterClient);
             assertEquals(newValue, newMasterValue);
-            slaveValue = executeForResult(Util.getReadAttributeOperation(base, attribute), slaveClient);
-            assertEquals(newValue, slaveValue);
-            serverValue = executeForResult(Util.getReadAttributeOperation(SERVER_ONE.append(base), attribute), masterClient);
-            assertEquals(newValue, serverValue);
+            if (expectSlaveEffect) {
+                ModelNode slaveValue = executeForResult(Util.getReadAttributeOperation(base, attribute), slaveClient);
+                assertEquals(newValue, slaveValue);
+                ModelNode serverValue = executeForResult(Util.getReadAttributeOperation(SERVER_ONE.append(base), attribute), masterClient);
+                assertEquals(newValue, serverValue);
+            } else {
+                executeForFailure(Util.getReadAttributeOperation(base, attribute), slaveClient);
+                executeForFailure(Util.getReadAttributeOperation(SERVER_ONE.append(base), attribute), masterClient);
+            }
         } catch (Exception | Error e) {
             caught = e;
             throw e;
@@ -244,6 +260,16 @@ public class RBACConfigTestCase {
             ModelNode response = client.execute(op);
             assertEquals(op.toString() + '\n' + response.toString(), SUCCESS, response.get(OUTCOME).asString());
             return response.get(RESULT);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private ModelNode executeForFailure(ModelNode op, ModelControllerClient client) {
+        try {
+            ModelNode response = client.execute(op);
+            assertEquals(op.toString() + '\n' + response.toString(), FAILED, response.get(OUTCOME).asString());
+            return response;
         } catch (IOException e) {
             throw new RuntimeException(e);
         }

--- a/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/RBACConfigTestCase.java
+++ b/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/RBACConfigTestCase.java
@@ -1,0 +1,251 @@
+/*
+Copyright 2016 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package org.jboss.as.test.integration.domain.mixed;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.ACCESS;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.APPLICATION_CLASSIFICATION;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.AUTHORIZATION;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.BASE_ROLE;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.CLASSIFICATION;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.CONFIGURED_APPLICATION;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.CONFIGURED_REQUIRES_WRITE;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.CONSTRAINT;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.CORE;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.CORE_SERVICE;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.HOST;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.HOSTS;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.HOST_SCOPED_ROLE;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.INCLUDE_ALL;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.MANAGEMENT;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OUTCOME;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.PROVIDER;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.RESULT;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.ROLE_MAPPING;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.RUNNING_SERVER;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SENSITIVITY_CLASSIFICATION;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SERVER_GROUPS;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SERVER_GROUP_SCOPED_ROLE;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUCCESS;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.TYPE;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.VAULT_EXPRESSION;
+import static org.junit.Assert.assertEquals;
+
+import java.io.IOException;
+
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.test.integration.domain.management.util.DomainTestSupport;
+import org.jboss.dmr.ModelNode;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Tests that RBAC configuration is properly handled in a mixed-domain.
+ *
+ * @author Brian Stansberry
+ */
+public class RBACConfigTestCase {
+
+    private static final PathAddress RBAC_BASE = PathAddress.pathAddress(PathElement.pathElement(CORE_SERVICE, MANAGEMENT),
+            PathElement.pathElement(ACCESS, AUTHORIZATION));
+    private static final PathAddress SERVER_ONE = PathAddress.pathAddress(PathElement.pathElement(HOST, "slave"),
+            PathElement.pathElement(RUNNING_SERVER, "server-one"));
+
+
+    private static ModelControllerClient masterClient;
+    private static ModelControllerClient slaveClient;
+
+    @Before
+    public void init() throws Exception {
+        DomainTestSupport support = KernelBehaviorTestSuite.getSupport(this.getClass());
+        masterClient = support.getDomainMasterLifecycleUtil().getDomainClient();
+        slaveClient = support.getDomainSlaveLifecycleUtil().getDomainClient();
+    }
+
+    @AfterClass
+    public synchronized static void afterClass() {
+        KernelBehaviorTestSuite.afterClass();
+        masterClient = slaveClient = null;
+    }
+
+    @Test
+    public void testWriteRBACProvider() throws IOException {
+        modifyTest(RBAC_BASE, PROVIDER);
+    }
+
+    @Test
+    public void testAddRoleMapping() throws IOException {
+        PathAddress address = RBAC_BASE.append(ROLE_MAPPING, "Operator");
+        String attribute = INCLUDE_ALL;
+        ModelNode value = new ModelNode(true);
+        ModelNode addOp = Util.createAddOperation(address);
+        addOp.get(attribute).set(value);
+
+        addTest(address, attribute, value, addOp);
+    }
+
+    @Test
+    public void testAddHostScopedRole() throws IOException {
+        PathAddress address = RBAC_BASE.append(HOST_SCOPED_ROLE, "WFCORE-1622_H");
+        String attribute = BASE_ROLE;
+        ModelNode value = new ModelNode("Operator");
+        ModelNode addOp = Util.createAddOperation(address);
+        addOp.get(attribute).set(value);
+        addOp.get(HOSTS).add("slave");
+
+        addTest(address, attribute, value, addOp);
+    }
+
+    @Test
+    public void testAddServerGroupScopedRole() throws IOException {
+        PathAddress address = RBAC_BASE.append(SERVER_GROUP_SCOPED_ROLE, "WFCORE-1622_S");
+        String attribute = BASE_ROLE;
+        ModelNode value = new ModelNode("Operator");
+        ModelNode addOp = Util.createAddOperation(address);
+        addOp.get(attribute).set(value);
+        addOp.get(SERVER_GROUPS).add("main-server-group");
+
+        addTest(address, attribute, value, addOp);
+    }
+
+    @Test
+    public void testModifySensitivityConstraint() throws IOException {
+
+        PathAddress mapping = RBAC_BASE.append(CONSTRAINT, SENSITIVITY_CLASSIFICATION)
+                .append(TYPE, CORE)
+                .append(CLASSIFICATION, "socket-config");
+        modifyTest(mapping, CONFIGURED_REQUIRES_WRITE, new ModelNode(true));
+    }
+
+    @Test
+    public void testModifyServerSSLSensitivityConstraint() throws IOException {
+
+        PathAddress mapping = RBAC_BASE.append(CONSTRAINT, SENSITIVITY_CLASSIFICATION)
+                .append(TYPE, CORE)
+                .append(CLASSIFICATION, "server-ssl");
+        modifyTest(mapping, CONFIGURED_REQUIRES_WRITE, new ModelNode(true));
+    }
+
+    @Test
+    public void testModifyApplicationConstraint() throws IOException {
+
+        PathAddress mapping = RBAC_BASE.append(CONSTRAINT, APPLICATION_CLASSIFICATION)
+                .append(TYPE, CORE)
+                .append(CLASSIFICATION, "deployment");
+        modifyTest(mapping, CONFIGURED_APPLICATION, new ModelNode(true));
+    }
+
+    @Test
+    public void testModifySensitiveExpressionsConstraint() throws IOException {
+
+        PathAddress mapping = RBAC_BASE.append(CONSTRAINT, VAULT_EXPRESSION);
+        modifyTest(mapping, CONFIGURED_REQUIRES_WRITE, new ModelNode(true));
+    }
+
+    private void modifyTest(PathAddress base, String attribute) throws IOException {
+        ModelNode masterValue = executeForResult(Util.getReadAttributeOperation(base, attribute), masterClient);
+        ModelNode slaveValue = executeForResult(Util.getReadAttributeOperation(base, attribute), slaveClient);
+        assertEquals(masterValue, slaveValue);
+        ModelNode serverValue = executeForResult(Util.getReadAttributeOperation(SERVER_ONE.append(base), attribute), masterClient);
+        assertEquals(masterValue, serverValue);
+
+        // Write the same value, so we don't need to clean up.
+        executeForResult(Util.getWriteAttributeOperation(base, attribute, masterValue), masterClient);
+
+        ModelNode newMasterValue = executeForResult(Util.getReadAttributeOperation(base, attribute), masterClient);
+        assertEquals(masterValue, newMasterValue);
+        slaveValue = executeForResult(Util.getReadAttributeOperation(base, attribute), slaveClient);
+        assertEquals(masterValue, slaveValue);
+        serverValue = executeForResult(Util.getReadAttributeOperation(SERVER_ONE.append(base), attribute), masterClient);
+        assertEquals(masterValue, serverValue);
+    }
+
+    private void modifyTest(PathAddress base, String attribute, ModelNode newValue) throws IOException {
+        ModelNode masterValue = executeForResult(Util.getReadAttributeOperation(base, attribute), masterClient);
+        ModelNode slaveValue = executeForResult(Util.getReadAttributeOperation(base, attribute), slaveClient);
+        assertEquals(masterValue, slaveValue);
+        ModelNode serverValue = executeForResult(Util.getReadAttributeOperation(SERVER_ONE.append(base), attribute), masterClient);
+        assertEquals(masterValue, serverValue);
+
+        // Write the same value, so we don't need to clean up.
+        Throwable caught = null;
+        executeForResult(Util.getWriteAttributeOperation(base, attribute, newValue), masterClient);
+        try {
+            ModelNode newMasterValue = executeForResult(Util.getReadAttributeOperation(base, attribute), masterClient);
+            assertEquals(newValue, newMasterValue);
+            slaveValue = executeForResult(Util.getReadAttributeOperation(base, attribute), slaveClient);
+            assertEquals(newValue, slaveValue);
+            serverValue = executeForResult(Util.getReadAttributeOperation(SERVER_ONE.append(base), attribute), masterClient);
+            assertEquals(newValue, serverValue);
+        } catch (Exception | Error e) {
+            caught = e;
+            throw e;
+        } finally {
+            if (!newValue.equals(masterValue)) {
+                try {
+                    executeForResult(Util.getWriteAttributeOperation(base, attribute, masterValue), masterClient);
+                } catch (RuntimeException e) {
+                    if (caught == null) {
+                        throw e;
+                    } else {
+                        e.printStackTrace(System.out);
+                    }
+                }
+            }
+        }
+
+    }
+
+    private void addTest(PathAddress address, String attribute, ModelNode value, ModelNode addOp) {
+        Throwable caught = null;
+        executeForResult(addOp, masterClient);
+        try {
+            ModelNode masterValue = executeForResult(Util.getReadAttributeOperation(address, attribute), masterClient);
+            assertEquals(value, masterValue);
+            ModelNode slaveValue = executeForResult(Util.getReadAttributeOperation(address, attribute), slaveClient);
+            assertEquals(value, slaveValue);
+            ModelNode serverValue = executeForResult(Util.getReadAttributeOperation(SERVER_ONE.append(address), attribute), masterClient);
+            assertEquals(value, serverValue);
+        } catch (Exception | Error e) {
+            caught = e;
+            throw e;
+        } finally {
+            try {
+                executeForResult(Util.createRemoveOperation(address), masterClient);
+            } catch (RuntimeException e) {
+                if (caught == null) {
+                    throw e;
+                } else {
+                    e.printStackTrace(System.out);
+                }
+            }
+        }
+    }
+
+    private ModelNode executeForResult(ModelNode op, ModelControllerClient client) {
+        try {
+            ModelNode response = client.execute(op);
+            assertEquals(op.toString() + '\n' + response.toString(), SUCCESS, response.get(OUTCOME).asString());
+            return response.get(RESULT);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+}

--- a/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap620/KernelBehavior620TestSuite.java
+++ b/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap620/KernelBehavior620TestSuite.java
@@ -33,7 +33,7 @@ import org.junit.runners.Suite;
  * @author Brian Stansberry
  */
 @RunWith(Suite.class)
-@Suite.SuiteClasses(value= {WildcardReads620TestCase.class})
+@Suite.SuiteClasses(value= {RBACConfig620TestCase.class, WildcardReads620TestCase.class})
 @Version(Version.AsVersion.EAP_6_2_0)
 public class KernelBehavior620TestSuite extends KernelBehaviorTestSuite {
 

--- a/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap620/RBACConfig620TestCase.java
+++ b/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap620/RBACConfig620TestCase.java
@@ -32,4 +32,9 @@ public class RBACConfig620TestCase extends RBACConfigTestCase {
     public static void beforeClass() {
         KernelBehavior620TestSuite.initializeDomain();
     }
+
+    @Override
+    protected boolean getSupportsServerSSL() {
+        return false;
+    }
 }

--- a/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap620/RBACConfig620TestCase.java
+++ b/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap620/RBACConfig620TestCase.java
@@ -1,0 +1,35 @@
+/*
+Copyright 2016 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package org.jboss.as.test.integration.domain.mixed.eap620;
+
+import org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase;
+import org.jboss.as.test.integration.domain.mixed.Version;
+import org.junit.BeforeClass;
+
+/**
+ * EAP 6.2 variant of RBACConfigTestCase.
+ *
+ * @author Brian Stansberry
+ */
+@Version(Version.AsVersion.EAP_6_2_0)
+public class RBACConfig620TestCase extends RBACConfigTestCase {
+
+    @BeforeClass
+    public static void beforeClass() {
+        KernelBehavior620TestSuite.initializeDomain();
+    }
+}

--- a/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap630/KernelBehavior630TestSuite.java
+++ b/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap630/KernelBehavior630TestSuite.java
@@ -24,8 +24,6 @@ package org.jboss.as.test.integration.domain.mixed.eap630;
 
 import org.jboss.as.test.integration.domain.mixed.KernelBehaviorTestSuite;
 import org.jboss.as.test.integration.domain.mixed.Version;
-import org.jboss.as.test.integration.domain.mixed.eap620.MixedDomain620TestSuite;
-import org.jboss.as.test.integration.domain.mixed.eap620.WildcardReads620TestCase;
 import org.junit.BeforeClass;
 import org.junit.runner.RunWith;
 import org.junit.runners.Suite;
@@ -35,7 +33,7 @@ import org.junit.runners.Suite;
  * @author Brian Stansberry
  */
 @RunWith(Suite.class)
-@Suite.SuiteClasses(value= {WildcardReads630TestCase.class})
+@Suite.SuiteClasses(value= {RBACConfig630TestCase.class, WildcardReads630TestCase.class})
 @Version(Version.AsVersion.EAP_6_3_0)
 public class KernelBehavior630TestSuite extends KernelBehaviorTestSuite {
 

--- a/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap630/RBACConfig630TestCase.java
+++ b/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap630/RBACConfig630TestCase.java
@@ -32,4 +32,9 @@ public class RBACConfig630TestCase extends RBACConfigTestCase {
     public static void beforeClass() {
         KernelBehavior630TestSuite.initializeDomain();
     }
+
+    @Override
+    protected boolean getSupportsServerSSL() {
+        return false;
+    }
 }

--- a/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap630/RBACConfig630TestCase.java
+++ b/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap630/RBACConfig630TestCase.java
@@ -1,0 +1,35 @@
+/*
+Copyright 2016 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package org.jboss.as.test.integration.domain.mixed.eap630;
+
+import org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase;
+import org.jboss.as.test.integration.domain.mixed.Version;
+import org.junit.BeforeClass;
+
+/**
+ * EAP 6.3 variant of RBACConfigTestCase.
+ *
+ * @author Brian Stansberry
+ */
+@Version(Version.AsVersion.EAP_6_3_0)
+public class RBACConfig630TestCase extends RBACConfigTestCase {
+
+    @BeforeClass
+    public static void beforeClass() {
+        KernelBehavior630TestSuite.initializeDomain();
+    }
+}

--- a/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap640/KernelBehavior640TestSuite.java
+++ b/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap640/KernelBehavior640TestSuite.java
@@ -33,7 +33,7 @@ import org.junit.runners.Suite;
  * @author Brian Stansberry
  */
 @RunWith(Suite.class)
-@Suite.SuiteClasses(value= {WildcardReads640TestCase.class})
+@Suite.SuiteClasses(value= {RBACConfig640TestCase.class, WildcardReads640TestCase.class})
 @Version(Version.AsVersion.EAP_6_4_0)
 public class KernelBehavior640TestSuite extends KernelBehaviorTestSuite {
 

--- a/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap640/RBACConfig640TestCase.java
+++ b/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap640/RBACConfig640TestCase.java
@@ -1,0 +1,35 @@
+/*
+Copyright 2016 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package org.jboss.as.test.integration.domain.mixed.eap640;
+
+import org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase;
+import org.jboss.as.test.integration.domain.mixed.Version;
+import org.junit.BeforeClass;
+
+/**
+ * EAP 6.4 variant of RBACConfigTestCase.
+ *
+ * @author Brian Stansberry
+ */
+@Version(Version.AsVersion.EAP_6_4_0)
+public class RBACConfig640TestCase extends RBACConfigTestCase {
+
+    @BeforeClass
+    public static void beforeClass() {
+        KernelBehavior640TestSuite.initializeDomain();
+    }
+}

--- a/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap640/RBACConfig640TestCase.java
+++ b/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap640/RBACConfig640TestCase.java
@@ -32,4 +32,9 @@ public class RBACConfig640TestCase extends RBACConfigTestCase {
     public static void beforeClass() {
         KernelBehavior640TestSuite.initializeDomain();
     }
+
+    @Override
+    protected boolean getSupportsServerSSL() {
+        return false;
+    }
 }

--- a/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap700/KernelBehavior700TestSuite.java
+++ b/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap700/KernelBehavior700TestSuite.java
@@ -33,7 +33,7 @@ import org.junit.runners.Suite;
  * @author Brian Stansberry
  */
 @RunWith(Suite.class)
-@Suite.SuiteClasses(value= {WildcardReads700TestCase.class})
+@Suite.SuiteClasses(value= {RBACConfig700TestCase.class, WildcardReads700TestCase.class})
 @Version(Version.AsVersion.EAP_7_0_0)
 public class KernelBehavior700TestSuite extends KernelBehaviorTestSuite {
 

--- a/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap700/RBACConfig700TestCase.java
+++ b/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap700/RBACConfig700TestCase.java
@@ -1,0 +1,35 @@
+/*
+Copyright 2016 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package org.jboss.as.test.integration.domain.mixed.eap700;
+
+import org.jboss.as.test.integration.domain.mixed.RBACConfigTestCase;
+import org.jboss.as.test.integration.domain.mixed.Version;
+import org.junit.BeforeClass;
+
+/**
+ * EAP 7.0 variant of RBACConfigTestCase.
+ *
+ * @author Brian Stansberry
+ */
+@Version(Version.AsVersion.EAP_7_0_0)
+public class RBACConfig700TestCase extends RBACConfigTestCase {
+
+    @BeforeClass
+    public static void beforeClass() {
+        KernelBehavior700TestSuite.initializeDomain();
+    }
+}

